lib/backend_common/backend_common/auth.py

```diff
@@ -184,7 +223,24 @@ def wrapper(*args, **kwargs):
             return 'Unauthorized', 401
         return wrapper
 
-    def require_scopes(self, scopes):
+    def _require_permissions(self, permissions):
+        if not self._require_login():
+            return False
+
+        with flask.current_app.app_context():
+            if not flask_login.current_user.has_permissions(permissions):
+                user = flask_login.current_user.get_id()
+                user_permissions = flask_login.current_user.get_permissions()
+                diff = ' OR '.join([
```

Comment:
Isn't there a function in taskcluster-client that does this ?

Reply:
yes, but we might not be using taskcluser for authorization. like we do in `shipit/api`.